### ISSUE TITLE
Newest version 4.0.1 broke our builds

### ISSUE THREAD

**Issue by bernienowel** [ReneSorensen/VSTSTools#17]
Error log:
```
##[debug]Tagging with '2.0.50-alpha'.
##[command] git tag   2.0.50-alpha -m NuGet Version 2.0.50-alpha
##[debug]Push tag to origin
##[command] git push origin 2.0.50-alpha
##[debug]Leaving C...\_tasks\GitTag_be6aa0ca-e62f-414d-aaa6-e9524b556482\4.0.1\tagging.ps1.
##[debug]Caught exception from task script.
##[debug]Error record:
##[debug]To https://<vsts-host>/_git/repo
##[debug]    + CategoryInfo          : NotSpecified: (:) [], RemoteException
##[debug]    + FullyQualifiedErrorId : RuntimeException
##[debug] 
##[debug]Script stack trace:
##[debug]
##[debug]Exception:
##[debug]System.Management.Automation.RemoteException: To https://<vsts-host>/_git/repo
##[error]To https://<vsts-host>/_git/repo
##[debug]Processed: ##vso[task.logissue type=error]To https://<vsts-host>/_git/repo
##[debug]Processed: ##vso[task.complete result=Failed]
##[section]Finishing: GitTag 
```
Potential fix in [tagging.ps1](https://github.com/ReneSorensen/VSTSTools/blob/master/GitTag/GitTag/tagging.ps1):
```
function showRedirectedOutput
{
    param ($output)

    $output | ForEach-Object -Process `
    {
        if ($_ -is [System.Management.Automation.ErrorRecord])
        {
            Write-Host "$StderrPrefix$_"
        }
        else
        {
            Write-Host "$_"
        }
    }
}
```

Wrap the calls where we're re-directing the stderr with a finally block so we can set an unset our ErrorActionPreferences. When writing the output from those errors we need to write them differently since they're of type ErrorRecord and the act of writing them out will throw an exception.

```
	try {
		Write-Verbose "Push tag to origin"
		Write-Host "##[command]"git push origin $tag
		$pushOutput =  git push origin $tag 2>&1

		showRedirectedOutput $tagOutput
		showRedirectedOutput $pushOutput
		
		if ($LastExitCode -ne 0) { 
			Write-Error "Something went wrong. Please check the logs."
		}
	}
	finally 
	{
		$script:ErrorActionPreference = $backupErrorActionPreference	
	}

```
This ignores that the output was being shown only in verbose mode when it was in error, but is functionally equivalent to the previous functionality.

**Comment by danquirk** [comment]
As mentioned in #16 I'm seeing the same thing.

**Comment by bernienowel** [comment]
Is there any chance you can re-publish the previous version while you work on a fix? I really like this extension but need to get our builds working again.

**Comment by VisualBean** [comment]
Sorry guys. - Im looking into fixing it.

**Comment by VisualBean** [comment]
A fix has been pushed #17 thanks @bernienowel.
